EIPS/eip-7742.md

```diff
@@ -33,67 +33,99 @@ This EIP also changes how the EL sources the current blob target value for two r
 
 ## Specification
 
-### Block structure and validity
+| Constant | Value | Remarks|
+| - | - | - |
+| `OLD_TARGET_BLOBS_PER_BLOCK` | `3` | blob target corresponding to [EIP-4844](./eip-4844.md) |
+| `TARGET_BLOBS_PER_BLOCK_EIP_7742` | `4` | increase the target blobs to `4`|
+| `MIN_BASE_FEE_PER_BLOB_GAS_EIP_7742` | `2**25` | the new min blob base fee |
+| `EXCESS_BLOB_GAS_NORMALIZATION_TARGET_BLOBS` | `128` | normalize excess gas to a target of `128` blobs |
+| `BLOB_BASE_FEE_UPDATE_FRACTION_PER_TARGET_BLOB` | `1112825` | used to calculate normalized update fraction for excess gas |
+| `BLOB_BASE_FEE_UPDATE_FRACTION_NORMALIZED` | `142441600`| excess gas update fraction for the normalized target of `128` blobs |
+
+### Block processing
 
 Upon activation of this EIP, execution clients **MUST** extend the header schema with an
-additional 64-bit field: the `target_blobs_per_block`. This value is set to the current target blob count. The Engine API
-is modified along with this EIP to provide the `target_blobs_per_block` with each payload and implementations can use this
-value to correctly set the block header field.
+additional 64-bit field: the `target_blobs_per_block`. This values is set to the one provided by the CL during block production.
 
-Validity of the `target_blobs_per_block` is guaranteed from the consensus layer, much like how withdrawals are handled.
+Any reference to `TARGET_BLOB_GAS_PER_BLOCK` from [EIP-4844](./eip-4844.md) (for e.g. for `excess_blob_gas` calculations) can be derived by taking the `target_blobs_per_block` from the EL block header and multiplying by `GAS_PER_BLOB`.
 
-When verifying a block, execution clients **MUST** ensure the target blob count in the block header matches the one
-provided by the consensus client.
+Validity of these values is guaranteed from the consensus layer, much like how withdrawals are handled. Hence when verifying a block, execution clients **MUST** ensure the target blobs per block in the block header matches the one provided by the consensus client.
 
-For a genesis block with no existing parent, the value should be set according to the agreed specification for the
-target blob count given by that genesis block's protocol rule set.
+Furthermore `get_base_fee_per_blob_gas` as specified by [EIP-4844](./eip-4844.md) is as modified below:
 
-### Block processing
 
-Upon activating this EIP (i.e. before processing any transactions),
-the verification of the blob maximum as given in EIP-4844 can be skipped. Concretely, this means any logic relating
-to `MAX_BLOB_GAS_PER_BLOCK` as given in EIP-4844 can be deprecated.
-Additionally, any reference to `TARGET_BLOB_GAS_PER_BLOCK` from EIP-4844 can be derived by taking the `target_blobs_per_block` from the CL and multiplying by `GAS_PER_BLOB` as given in EIP-4844.
+```python
+def calc_excess_blob_gas(parent: Header) -> int:
+  # normalize parent's excess blob gas if this block was fork block
+  if(parent.timestamp < FORK_TIMESTAMP)
```

Comment:
there is no way to deploy this rule before the fork and so this clause for `parent.timestamp < FORK_TIMESTAMP` would never be reachable, it should be removed

Reply:
if you want to do the conversion logic at the fork boundary, I think this should instead be `if parent.timestamp == FORK_TIMESTAMP`

Reply:
At the fork boundary, I think it's possible for `parent.timestamp < FORK_TIMESTAMP` and `header.timestamp >= FORK_TIMESTAMP`.
This new calculation is activated at the fork but still operating on the parent block's data (which doesn't include target_blobs_per_block) so I think it makes sense to use the old hardcoded target here.

Reply:
ahh yes as @siladu pointed out, parent is pre 7742 and this logic runs for calculation of the excess gas for the current fork block

Reply:
ah, I see what's going on now, thanks!